src/common.rs

```diff
@@ -0,0 +1,11 @@
+use pyo3::Python;
+
+/// Result type used in the crate that holds a String as the Err type.
+pub(crate) type StringResult<T> = Result<T, String>;
+
+// This is a trait that can be implemented by any parent class.
+// The function returns one of the possible subclasses instances.
+pub(crate) trait ParentClass {
```

Comment:
Maybe this should be in `pyo3_extensions`? It's pretty specifically an extension to the way we're using PyO3 Pyclasses.